File: compress/compress.go
package main

import (
	"bytes"
	"compress/zlib"
	"encoding/base64"
	"fmt"
	"io"
)

func encode(content string) string {
	src := []byte(content)
	var in bytes.Buffer
	w := zlib.NewWriter(&in)
	w.Write(src)
	w.Close()
	return base64.StdEncoding.EncodeToString(in.Bytes())
}

func decode(s string) string {
	data, err := base64.StdEncoding.DecodeString(s)
	if err != nil {
		return ""
	}
	var out bytes.Buffer
	r, _ := zlib.NewReader(bytes.NewReader(data))
	io.Copy(&out, r)
	return out.String()
}

func main() {
	s := "123123123"
	content := ""
	for i := 0; i < 100; i++ {
		content += s
	}
	fmt.Println(content)
	fmt.Println(decode(encode(content)))
	fmt.Println(decode("eJwzNDI2HEWjaBQNHAIAaLuvyQ=="))
}

File: compress/compress.py
import base64
import zlib

def encode(content):
    return base64.b64encode(zlib.compress(content))

def decode(s):
    return zlib.decompress(base64.b64decode(s))

if __name__ == "__main__":
    raw = b"123123123" * 100
    print(len(raw))
    enc = encode(raw)
    print(len(enc))
    print(enc)
    dec = decode(enc)
    print(dec)
    assert raw == dec
